### PR TITLE
test: Add test for --overlay-* path=value

### PR DESCRIPTION
**Pull request by mkmik**
Tests for: https://github.com/kubecfg/kubecfg/pull/303

### CODE REVIEW

File: pkg/kubecfg/vm_test.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	obj := objs[0]
+	t.Logf("obj: %q", obj)
+
+	if got, want := obj.GetObjectKind().GroupVersionKind().Kind, "ConfigMap"; got != want {
+		t.Fatalf("got: %q, want: %q", got, want)
+	}
+	return obj.GetAnnotations()
+}
+
+func TestReadObjec(t *testing.T) {
```
Comment (jdockerty) [inline pkg/kubecfg/vm_test.go:56]:
```suggestion
func TestReadObject(t *testing.T) {
```

File: pkg/kubecfg/vm_test.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	if _, err := io.WriteString(tmpfile, body); err != nil {
+		tmpfile.Close()
+		t.Fatal(err)
+	}
+	if err := tmpfile.Close(); err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(func() {
+		os.Remove(tmpfile.Name())
+	})
```
Comment (jdockerty) [inline pkg/kubecfg/vm_test.go:28]:
```suggestion
	tmpfile, err := os.CreateTemp(t.TempDir(), "kubecfg-test-*.jsonnet")
	if err != nil {
		t.Fatal(err)
	}
	if _, err := io.WriteString(tmpfile, body); err != nil {
		tmpfile.Close()
		t.Fatal(err)
	}
	if err := tmpfile.Close(); err != nil {
		t.Fatal(err)
	}
```

Thoughts on doing something like this?

I.e. we pass in [`t.TempDir()`](https://pkg.go.dev/testing#T.TempDir) which is from the `testing` stdlib that will automatically provide a directory that will be cleaned up after the test anyway - allowing us to slightly simplify the setup and remove our own call to `t.Cleanup`.


> TempDir returns a temporary directory for the test to use. The directory is automatically removed by Cleanup when the test and all its subtests complete.

**Review by jdockerty — CHANGES_REQUESTED** [review]
Mainly for the typo :smile: 

Otherwise LGTM :rocket:

File: pkg/kubecfg/vm_test.go (hunk 2 above)
Comment (mkmik) [reply pkg/kubecfg/vm_test.go:28]: Ah yeah I forgot about that helper

**Comment by mkmik** [comment]
suggestions accepted; PTAL

**Review by jdockerty — APPROVED** [review]

**Pull request merged**